src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/TOV.cpp

```diff
@@ -0,0 +1,183 @@
+
+      dudh = -2.0 * u * (1.0 - 2.0 * v) / (4.0 * M_PI * u * get(pressure) + v);
+
+      dvdh = -(1.0 - 2.0 * v) * (4.0 * M_PI * u * get(energy_density) - v) /
+             (4.0 * M_PI * u * get(pressure) + v);
+    }
+  }
+
+  else {
+    return;
+  }
+}
+
+template <bool IsRelativistic, size_t dim>
+InterpolationOutput TOV_Output::tov_solver(
+    std::unique_ptr<EquationsOfState::EquationOfState<IsRelativistic, dim>>&
+        polyM,
+    double central_mass_density_in) noexcept {
+  state_type u_and_v = {0.0, 0.0};
+
+  Scalar<double> central_mass_density{central_mass_density_in};
+
+  double hc = std::log(
+      get(polyM->specific_enthalpy_from_density(central_mass_density)));
+
+  typedef boost::numeric::odeint::runge_kutta_dopri5<state_type> dopri5_type;
+  typedef boost::numeric::odeint::controlled_runge_kutta<dopri5_type>
+      controlled_dopri5_type;
+  typedef boost::numeric::odeint::dense_output_runge_kutta<
+      controlled_dopri5_type>
+      dense_output_dopri5_type;
+
+  dense_output_dopri5_type dopri5 =
+      make_dense_output(1.0e-14, 1.0e-20, dopri5_type());
+
+  Observer observer{};
+
+  boost::numeric::odeint::integrate_adaptive(
+      dopri5,
+      [&polyM](state_type& lindblom_u_and_v, state_type& lindblom_dudh_and_dvdh,
+               double lindblom_enthalpy) {
+        return lindblom(lindblom_u_and_v, lindblom_dudh_and_dvdh,
+                        lindblom_enthalpy, polyM);
+      },
+      u_and_v, hc, 0.000, -1.0e-10, std::ref(observer));
+
+  InterpolationOutput interout(observer.radius, observer.mass,
+                               observer.log_enthalpy);
+
+  return interout;
+}
+
+template <bool IsRelativistic, size_t dim>
+InterpolationOutput TOV_Output::tov_solver_for_testing(
+    std::unique_ptr<EquationsOfState::EquationOfState<IsRelativistic, dim>>&
+        polyM,
+    double central_mass_density_in, double h_final) noexcept {
+  state_type u_and_v = {0.0, 0.0};
+
+  Scalar<double> central_mass_density{central_mass_density_in};
+
+  double hc = std::log(
+      get(polyM->specific_enthalpy_from_density(central_mass_density)));
+
+  typedef boost::numeric::odeint::runge_kutta_dopri5<state_type> dopri5_type;
+  typedef boost::numeric::odeint::controlled_runge_kutta<dopri5_type>
+      controlled_dopri5_type;
+  typedef boost::numeric::odeint::dense_output_runge_kutta<
+      controlled_dopri5_type>
+      dense_output_dopri5_type;
+
+  dense_output_dopri5_type dopri5 =
+      make_dense_output(1.0e-14, 1.0e-20, dopri5_type());
+
+  Observer observer{};
+
+  boost::numeric::odeint::integrate_adaptive(
+      dopri5,
+      [&polyM](state_type& lindblom_u_and_v, state_type& lindblom_dudh_and_dvdh,
+               double lindblom_enthalpy) {
+        return lindblom(lindblom_u_and_v, lindblom_dudh_and_dvdh,
+                        lindblom_enthalpy, polyM);
+      },
+      u_and_v, hc, h_final, -1.0e-10, std::ref(observer));
```

Comment:
inline hc here;  what is -1.0e-10?  can you compute it from the dudh and dvdh above?